### DISCUSSION TITLE
[Refit] What's the reason for generated HttpClient name to include also assembly full name?

### DISCUSSION THREAD

**Discussion by yohny** [reactiveui/refit#1464] [Q&A]
Currently the generated name for Refit `HttpClient` also contains assembly full name as visible here: https://github.com/reactiveui/refit/blob/246ee8d9989c29092fdaba6821d5b3098a5ccf9e/Refit/UniqueName.cs#L49
This results in name like `System.Net.Http.HttpClient.Refit.Implementation.Generated+SerilogRefiReproControllersIApi, SerilogRefiRepro, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null` which makes name long a bloated in scenarios like logging - see the issue [here](https://github.com/serilog/serilog/issues/1793).
Even without the assembly full name, the generated name would be unique in cases when the refit interface with the same name would be also defined in another assembly, as the namespace that is already included right after `+` would be different.
Would it be possible to get rid of the assembly name?

**Comment by glennawatson** [comment]
I'd be inclined to keep the current logging functionality as is. It has more flexibility for existing users who rely on the full name.

**Comment by yohny** [comment]
What would be a scenario where someone would rely on specific generated httpClient name?
I can see it used as logging context, where you dont rely on a specific value, its just information where the log comes from. You also dont explicitly ask for refit httpClient instance througth DI using generated client name, as there is currently no way of getting it, instead you would use the refit interface. I'm I missing something?
